File: backend/app/schemas/character.py
"""Character API request/response schemas - Simplified for file system based storage."""

from typing import List
from pydantic import BaseModel, Field

from app.models.character import UserCharacter


class CreateCharacterRequest(BaseModel):
    """Request to create a new character."""
    name: str = Field(..., min_length=1, max_length=100, description="Character name")
    prompt: str = Field(..., min_length=1, description="System prompt for the character")


class UpdateCharacterPromptRequest(BaseModel):
    """Request to update a character's prompt."""
    prompt: str = Field(..., min_length=1, description="New system prompt")


class UserCharacterListResponse(BaseModel):
    """Response with list of user characters."""
    characters: List[UserCharacter]
    count: int


class UserCharacterResponse(BaseModel):
    """Response with a single user character."""
    character: UserCharacter

File: frontend/src/types/character.ts
/** Character type definitions - Simplified for file system based storage */

// User-created character - Main character model
// character_id IS the UUID4
export interface Character {
  character_id: string;  // UUID4
  name: string;
  created_at: string;
  updated_at: string;
}

// Backward compatibility alias
export type UserCharacter = Character;
